File: Front-end/app/common/service/testResult.service.js
export default function testResultService($http) {
    var service = this;

    service.getTestResultsByEmployeeIdAndTestPeriodId = function (employeeId, testPeriodId) {
        return $http({
            method: "GET",
            url: EM_REST_SERVER_URI + "/employees/" + employeeId + "/reports/" + testPeriodId + "/results"
        }).then(function (response) {
            return response.data;
        });
    };

    service.sendTestAnswers = function (employeeId, testPeriodId, testId, answers) {
        return $http({
            method: "POST",
            url: EM_REST_SERVER_URI + "/employees/" + employeeId + "/reports/" + testPeriodId + "/results",
            data: {
                testId: testId,
                answers: answers
            }
        }).then(function (response) {
            return response.data;
        });
    };
}

File: Back-end/src/main/java/by/bsuir/em/service/TestResultService.java
package by.bsuir.em.service;

import by.bsuir.em.dto.TestResultDto;

import java.util.List;

public interface TestResultService {
    List<TestResultDto> getTestResultsByEmployeeIdAndTestPeriodId(Long employeeId, Long testPeriodId);
    TestResultDto addTestResult(TestResultDto testResultDto);
}
